src/components/transactions/FilterModal.tsx

```diff
@@ -0,0 +1,393 @@
+                isSelected={filterState.inProgressMinDateIsSet}
+                onPress={() => {
+                  if (maxDatePickerIsVisible) {
+                    setMaxDatePickerIsVisible(false);
+                  }
+                  setMinDatePickerIsVisible(true);
+                }}
+                icon="calendar-today"
+              >
+                <Body1Text>
+                  {filterState.inProgressMinDateIsSet ? (
+                    moment(filterState.inProgressMinDate).format('M/DD/YYYY')
+                  ) : (
+                    <MidGrayText>From</MidGrayText>
+                  )}
+                </Body1Text>
+              </FilterField>
+            </LeftAlignContainer>
+            <HorizontalSpacing />
+            <RightAlignContainer>
+              <FilterField
+                isSelected={filterState.inProgressMaxDateIsSet}
+                onPress={() => {
+                  if (minDatePickerIsVisible) {
+                    setMinDatePickerIsVisible(false);
+                  }
+                  setMaxDatePickerIsVisible(true);
+                }}
+                icon="calendar-today"
+              >
+                <Body1Text>
+                  {filterState.inProgressMaxDateIsSet ? (
+                    moment(filterState.inProgressMaxDate).format('M/D/YY')
+                  ) : (
+                    <MidGrayText>To</MidGrayText>
+                  )}
+                </Body1Text>
+              </FilterField>
+            </RightAlignContainer>
+            {minDatePickerIsVisible && (
+              <DatePickerContainer>
+                <RNDateTimePicker
+                  mode="date"
+                  value={filterState.inProgressMinDate}
+                  display={Platform.OS === 'ios' ? 'inline' : 'default'}
+                  onChange={e => {
+                    setMinDatePickerIsVisible(false);
+                    if (e.type === 'set' && e.nativeEvent.timestamp) {
+                      filterDispatch({
+                        type: 'SET_MIN_DATE',
+                        date: new Date(e.nativeEvent.timestamp),
+                      });
+                    }
+                  }}
+                />
+              </DatePickerContainer>
+            )}
+
+            {maxDatePickerIsVisible && (
+              <DatePickerContainer>
+                <RNDateTimePicker
+                  mode="date"
+                  value={filterState.inProgressMaxDate}
+                  display={Platform.OS === 'ios' ? 'inline' : 'default'}
+                  onChange={e => {
+                    setMaxDatePickerIsVisible(false);
+                    if (e.type === 'set' && e.nativeEvent.timestamp) {
+                      filterDispatch({
+                        type: 'SET_MAX_DATE',
+                        date: new Date(e.nativeEvent.timestamp),
+                      });
+                    }
+                  }}
+                />
+              </DatePickerContainer>
```

Comment:
is there any way to abstract away minDatePicker and maxDatePicker into a separate component? because there's a lot of repetition